lisa/sut_orchestrator/azure/common.py

```diff
@@ -145,11 +145,12 @@ class AzureNodeSchema:
     # image.
     is_linux: Optional[bool] = None
 
-    _marketplace: InitVar[Optional[AzureVmMarketplaceSchema]] = None
+    def __post_init__(self) -> None:
```

Comment:
The `__post_init__` may not be called, when creating the object in code. I'm not sure the logic, but there was some reason to declare them like InitVar, instead of put in `__post_init__`.

Reply:
Please try if it works with an Azure run.

Reply:
`__post_init__()` is called by the `__init__()` method generating the data classes. This is where instance variables like this should go. `InitVar` types are passed to `__post_init__()`, but there was no logic for them, so that implementation was incomplete. It looks like they were just done that way to make mypy happy, but the logic in mypy changed and it was causing errors.

Kicked of an azure test local and it ran fine